patches/cip-core/v2-0001-Add-option-to-use-swupdate-handler-roundrobin.patch

```diff
@@ -0,0 +1,182 @@
+ KFEATURE_luahandler = ""
+ KFEATURE_luahandler[KCONFIG_SNIPPETS] = "file://swupdate_defconfig_luahandler.snippet"
+-KFEATURE_luahandler[SRC_URI] = "file://${SWUPDATE_LUASCRIPT}"
++KFEATURE_luahandler[SRC_URI] = "${@ 'file://${SWUPDATE_LUASCRIPT}' \
++                                  if d.getVar('SWUPDATE_USE_ROUND_ROBIN_HANDLER_REPO') == '0' else '' }"
+ 
+ KFEATURE_DEPS = ""
+ KFEATURE_DEPS[luahandler] = "lua"
+@@ -57,8 +65,6 @@ KFEATURE_u-boot[DEBIAN_DEPENDS] = "${@ 'libubootenv0.1, u-boot-${MACHINE}-config
+ KFEATURE_u-boot[DEPENDS] = "libubootenv"
+ KFEATURE_u-boot[KCONFIG_SNIPPETS] = "file://swupdate_defconfig_u-boot.snippet"
+ 
+-SWUPDATE_LUASCRIPT ?= "swupdate_handlers.lua"
+-
+ def get_bootloader_featureset(d):
+     bootloader = d.getVar("SWUPDATE_BOOTLOADER", True) or ""
+     if bootloader == "efibootguard":
+diff --git a/kas/opt/ebg-secure-boot-base.yml b/kas/opt/ebg-secure-boot-base.yml
+index 30ca35a..484d0e5 100644
+--- a/kas/opt/ebg-secure-boot-base.yml
++++ b/kas/opt/ebg-secure-boot-base.yml
+@@ -16,3 +16,6 @@ local_conf_header:
+   initramfs: |
+     IMAGE_INSTALL += "initramfs-abrootfs-secureboot"
+     SWU_DESCRIPTION = "secureboot"
++  swupdate-secureboot: |
++    SWUPDATE_USE_ROUND_ROBIN_HANDLER_REPO = "0"
++    SWUPDATE_LUASCRIPT = "swupdate_handler.efibootguard.secureboot.lua"
+diff --git a/recipes-core/images/files/secure-boot/sw-description.tmpl b/recipes-core/images/files/secure-boot/sw-description.tmpl
+index bce97d0..897d819 100644
+--- a/recipes-core/images/files/secure-boot/sw-description.tmpl
++++ b/recipes-core/images/files/secure-boot/sw-description.tmpl
+@@ -16,7 +16,7 @@ software =
+             filename = "${ROOTFS_PARTITION_NAME}";
+             device = "fedcba98-7654-3210-cafe-5e0710000001,fedcba98-7654-3210-cafe-5e0710000002";
+             type = "roundrobin";
+-            compressed = "true";
++            compressed = "zlib";
+             filesystem = "ext4";
+     });
+     files: ({
+diff --git a/recipes-core/images/files/sw-description.tmpl b/recipes-core/images/files/sw-description.tmpl
+index bb34088..3309271 100644
+--- a/recipes-core/images/files/sw-description.tmpl
++++ b/recipes-core/images/files/sw-description.tmpl
+@@ -16,21 +16,30 @@ software =
+             filename = "${ROOTFS_PARTITION_NAME}";
+             device = "fedcba98-7654-3210-cafe-5e0710000001,fedcba98-7654-3210-cafe-5e0710000002";
+             type = "roundrobin";
+-            compressed = "true";
++            compressed = "zlib";
+             filesystem = "ext4";
++            properties: {
++                        subtype = "image";
++            };
+     });
+     files: ({
+             filename = "${KERNEL_IMAGE}";
+             path = "vmlinuz";
+-            type = "kernelfile";
+-            device = "sda2,sda3";
++            type = "roundrobin";
++            device = "fedcba98-7654-3210-cafe-5e0710000001->sda2,fedcba98-7654-3210-cafe-5e0710000002->sda3";
```

Comment:
Are you sure these device nodes `/dev/sda?` exist on the IoT2050?

Reply:
/dev/mmcblk0 or 1, depending on whether we are running from eMMC or SD card. USB drivers would be possible as well, then sda & Co. Can we avoid such device names?

Reply:
If we drop `regen-rootfs-uuid` we can use the partuuids.

Reply:
regen-rootfs-uuid is for the scenario of using the same image on the device as well as on a installation or recovery sticks. If we remove it, we need different images or even approaches (install-on-emmc won't work anymore).

If I get it right, we will need a stable partuuid, set on the device and also known to the build system when generating SWUs, correct?

Reply:
Y

> regen-rootfs-uuid is for the scenario of using the same image on the device as well as on a installation or recovery sticks. If we remove it, we need different images or even approaches (install-on-emmc won't work anymore).
> 
> If I get it right, we will need a stable partuuid, set on the device and also known to the build system when generating SWUs, correct?

Yes we need a stable partuuid known during the build. The device in the sw-description can currently contain device like `sda1` or partuuids.  With [swupdate-handler-roundrobin issue 9](https://gitlab.com/cip-playground/swupdate-handler-roundrobin/-/issues/9) solved we could also use part labels.